From 000d48992bc7b9bfc8956f5393dcc83d0126a801 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Sat, 29 Jun 2019 10:55:32 +0800
Subject: [PATCH] kickstart: not support Authselect/Firewall/Firstboot

We do not support kickstart for now

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

Refactor and update patch context for version anaconda 32.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 anaconda.py                | 8 --------
 pyanaconda/installation.py | 8 +-------
 pyanaconda/kickstart.py    | 5 -----
 3 files changed, 1 insertion(+), 20 deletions(-)

diff --git a/anaconda.py b/anaconda.py
index 4c4c9176f0..9f799d9d35 100755
--- a/anaconda.py
+++ b/anaconda.py
@@ -536,14 +536,6 @@ if __name__ == "__main__":
     if anaconda.methodstr and not ksdata.method.seen:
         startup_utils.set_installation_method_from_anaconda_options(anaconda, ksdata)
 
-    from pyanaconda.modules.common.constants.services import SECURITY
-    security_proxy = SECURITY.get_proxy()
-    # Override the selinux state from kickstart if set on the command line
-    if conf.security.selinux != constants.SELINUX_DEFAULT:
-        security_proxy.SetSELinux(conf.security.selinux)
-    # Enable fingerprint option by default (#481273).
-    if not flags.automatedInstall:
-        security_proxy.SetFingerprintAuthEnabled(True)
 
     from pyanaconda import localization
     # Set the language before loading an interface, when it may be too late.
diff --git a/pyanaconda/installation.py b/pyanaconda/installation.py
index e8b147bb62..4f8f1295f1 100644
--- a/pyanaconda/installation.py
+++ b/pyanaconda/installation.py
@@ -82,9 +82,6 @@ def _prepare_configuration(payload, ksdata):
     os_config = TaskQueue("Installed system configuration", N_("Configuring installed system"))
 
     # add installation tasks for the Security DBus module
-    security_proxy = SECURITY.get_proxy()
-    security_dbus_tasks = security_proxy.InstallWithTasks()
-    os_config.append_dbus_tasks(SECURITY, security_dbus_tasks)
 
     # add installation tasks for the Services DBus module
     services_proxy = SERVICES.get_proxy()
@@ -102,9 +99,7 @@ def _prepare_configuration(payload, ksdata):
     os_config.append_dbus_tasks(LOCALIZATION, localization_dbus_tasks)
 
     # add the Firewall configuration task
-    firewall_proxy = NETWORK.get_proxy(FIREWALL)
-    firewall_dbus_task = firewall_proxy.InstallWithTask()
-    os_config.append_dbus_tasks(NETWORK, [firewall_dbus_task])
+
 
     configuration_queue.append(os_config)
 
@@ -159,7 +154,6 @@ def _prepare_configuration(payload, ksdata):
 
     # realm join
     # - this can run only after network is configured in the target system chroot
-    configuration_queue.append_dbus_tasks(SECURITY, [security_proxy.JoinRealmWithTask()])
 
     post_scripts = TaskQueue("Post installation scripts", N_("Running post-installation scripts"))
     post_scripts.append(Task("Run post installation scripts", runPostScripts, (ksdata.scripts,)))
diff --git a/pyanaconda/kickstart.py b/pyanaconda/kickstart.py
index 7dbf83971a..b7c0807dfd 100644
--- a/pyanaconda/kickstart.py
+++ b/pyanaconda/kickstart.py
@@ -348,17 +348,12 @@ class AnacondaSection(Section):
 # This is just the latest entry from pykickstart.handlers.control with all the
 # classes we're overriding in place of the defaults.
 commandMap = {
-    "auth": UselessCommand,
-    "authconfig": UselessCommand,
-    "authselect": UselessCommand,
     "autopart": UselessCommand,
     "btrfs": UselessCommand,
     "bootloader": UselessCommand,
     "clearpart": UselessCommand,
     "eula": Eula,
     "fcoe": UselessCommand,
-    "firewall": UselessCommand,
-    "firstboot": UselessCommand,
     "group" : UselessCommand,
     "ignoredisk": UselessCommand,
     "iscsi": UselessCommand,
-- 
2.7.4

